validator/src/main.rs

```diff
@@ -190,17 +193,42 @@ fn initialize_ledger_path(
                 None
             }
         })
-        .next()
-        .unwrap_or_else(|| {
-            error!(
-                "Entrypoint ({:?}) is not running the RPC service",
-                entrypoint.gossip
-            );
-            exit(1);
-        });
+        .next();
 
-    let client = RpcClient::new_socket(rpc_addr);
-    let genesis_hash = client.get_genesis_hash().map_err(|err| err.to_string())?;
+    if let Some(rpc_addr) = rpc_addr {
+        Ok((rpc_addr, RpcClient::new_socket(rpc_addr)))
+    } else {
+        Err(format!(
+            "No node including entrypoint ({:?}) is running the RPC service",
+            entrypoint.gossip
+        ))
+    }
+}
+
+fn check_vote_account(rpc_client: &RpcClient, vote_account: &Pubkey) -> Result<(), String> {
+    let found_vote_account = rpc_client
+        .get_account(vote_account)
+        .map_err(|err| err.to_string())?;
+
+    if found_vote_account.owner == solana_vote_api::id() {
+        Ok(())
```

Comment:
Let's add additional checks in here for these two vote account fields:
https://github.com/solana-labs/solana/blob/f54cfcdb8f2c6f44f80c1c7b6ec02fb2e71fa3e3/programs/vote_api/src/vote_state.rs#L87-L89

Reply:
Done. Maybe is this something you wanted to check? https://github.com/solana-labs/solana/pull/6790/files#diff-1f422d7e73618aca7bce9935356de012R223